cabal-install/src/Distribution/Client/ProjectPlanning/SetupPolicy.hs

```diff
@@ -158,6 +165,19 @@ mkDefaultSetupDeps compiler platform pkg =
     csvToVersion :: CabalSpecVersion -> Version
     csvToVersion = mkVersion . cabalSpecMinimumLibraryVersion
 
+-- | Returns an implicit dependency on @hooks-exe@ needed to create a
+-- @Setup.hs@ executable from a 'SetupHooks' value, if @build-type: Hooks@.
+--
+-- @since 3.14.0.0
+mkHooksSetupImplicitDeps
+  :: PackageDescription
+  -> Maybe [Dependency]
+mkHooksSetupImplicitDeps pkg
+  | Hooks <- buildType pkg =
+      Just [Dependency hooksExePkgname anyVersion mainLibSet]
```

Comment:
Maybe we should extend this with `base` and `Cabal`, since these won't be implicitly included for existing `setup-depends` (as mentioned in my previous comment, this is caused by the setup-depend list is explicit).

Reply:
I'm extending it with `Cabal` now.